src/main/java/org/springframework/data/couchbase/config/AbstractCouchbaseConfiguration.java

```diff
@@ -273,6 +287,31 @@ public CouchbaseMappingContext couchbaseMappingContext(CustomConversions customC
 		return mappingContext;
 	}
 
+	/**
+	 * Creates a {@link Module} ObjectMapper mapper = new ObjectMapper();
+	 * mapper.registerModule(couchbaseJsonValueModule()); <----- ClusterEnvironment env = ClusterEnvironment.builder()
+	 * .jsonSerializer(new JacksonJsonSerializer(mapper)) .build();
+	 *
+	 * @throws Exception on Bean construction failure.
+	 * @return
+	 */
+	@Bean
+	public com.couchbase.client.core.deps.com.fasterxml.jackson.databind.Module couchbaseJsonValueModule() {
+		return new RepackagedJsonValueModule();
+	}
+
+	@Bean
+	public ObjectMapper couchbaseObjectMapper() {
+		ObjectMapper mapper = new ObjectMapper();
+		mapper.configure(com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+		mapper.registerModule(new JsonValueModule());
+		CryptoManager cryptoManager = null;
+		if (cryptoManager != null) {
+			mapper.registerModule(new EncryptionModule(cryptoManager));
+		}
+		return mapper;
+	}
+
```

Comment:
the couchbaseObjectMapper that gets used and exposed. Created the same as in ClusterEnvironment.Builder.